src/backend/distributed/utils/citus_clauses.c

```diff
@@ -211,7 +222,7 @@ citus_evaluate_expr(Expr *expr, Oid result_type, int32 result_typmod,
 	{
 		planState = masterEvaluationContext->planState;
 
-		if (IsA(expr, Param))
+		if (FindNodeCheck((Node *) expr, IsParam))
```

Comment:
@marcocitus please look into the first commit as well. That is actually a more accurate approach, because we should find the `PARAM` while traversing `Function`.

However, this second commit seems easier to read, and I am undecided on which to pick

Reply:
Does this mean we would evaluate an expression like `f($1+random())` in full, or only the parameter?